scheds/rust/scx_layered/src/main.rs

```diff
@@ -875,12 +875,23 @@ impl CpuPool {
         }
     }
 
-    fn alloc_cpus<'a>(&'a mut self, cpus: &BitVec) -> Option<&'a BitVec> {
-        let cores = self.cpus_to_cores(cpus).ok()?;
-        for core in cores.iter_ones() {
-            self.available_cores.set(core, false);
-            self.update_fallback_cpu();
-            return Some(&self.core_cpus[core]);
+    fn alloc_cpus<'a>(&'a mut self, layer: &Layer) -> Option<&'a BitVec> {
+        let available_cpus = self.available_cpus_in_mask(&layer.allowed_cpus);
+        let available_cores = self.cpus_to_cores(&available_cpus).ok()?;
+
+        for alloc_core in layer.core_alloc_order() {
+            match available_cores.get(*alloc_core) {
```

Comment:
this goes from the core id to the bitmask bit for checking if the core is allowed to be used